subsys/nrf_rpc/rpc_utils/server/dev_info.c

```diff
@@ -6,10 +6,205 @@
 
 #include <ncs_commit.h>
 #include <nrf_rpc/nrf_rpc_serialize.h>
+#include <nrf_rpc/rpc_utils/dev_info.h>
 #include <rpc_utils_group.h>
+#include <zephyr/debug/coredump.h>
+#include <zephyr/drivers/coredump.h>
+#include <zephyr/random/random.h>
 
 #include <nrf_rpc_cbor.h>
 
+#define DUMP_META_VERSION 1
+#define ASSERT_FILENAME_SIZE 255
+
+#define DUMP_METADATA_SIZE(meta) \
+	sizeof((meta).uuid) + 1 + \
+	sizeof((meta).reason) + 1 + \
+	sizeof((meta).lr) + 1 + \
+	sizeof((meta).pc) + 1 + \
+	sizeof((meta).xpsr) + 1 + \
+	sizeof((meta).sp) + 1 + \
+	sizeof((meta).assert_line) + 1 + \
+	(meta.assert_line ? (strlen(meta.assert_filename) + 3) : 1)
+
+struct arm_arch_block {
+	struct {
+		uint32_t r0;
+		uint32_t r1;
+		uint32_t r2;
+		uint32_t r3;
+		uint32_t r12;
+		uint32_t lr;
+		uint32_t pc;
+		uint32_t xpsr;
+		uint32_t sp;
+
+		/* callee registers - optionally collected in V2 */
+		uint32_t r4;
+		uint32_t r5;
+		uint32_t r6;
+		uint32_t r7;
+		uint32_t r8;
+		uint32_t r9;
+		uint32_t r10;
+		uint32_t r11;
+	} r;
+} __packed;
+
+struct dump_metadata
+{
+	struct dump_metadata_header {
+		uint8_t magic[8];
+		uint32_t uuid;
+		uint8_t version;
+		uint8_t has_assert_info;
+	} __packed header;
+	uint32_t assert_line;
+	const char assert_filename[ASSERT_FILENAME_SIZE];
+} __packed;
+
+static struct dump_metadata next_dump_info;
+
+static struct coredump_mem_region_node dump_region = {
+	.start = (uintptr_t)&next_dump_info,
+	.size = sizeof(next_dump_info.header)
+};
+
+static const uint8_t CRASH_INFO_MAGIC[] = { 'D', 'U', 'M', 'P', 'I', 'N', 'F', 'O' };
+
+static int load_coredump_fragment(uint16_t offset, void *buffer, size_t length)
+{
+	struct coredump_cmd_copy_arg copy_params;
+
+	copy_params.offset = offset;
+	copy_params.buffer = (uint8_t *)buffer;
+	copy_params.length = length;
+
+	return coredump_cmd(COREDUMP_CMD_COPY_STORED_DUMP, &copy_params);
+}
+
+static int load_crash_info(struct nrf_rpc_crash_info *info)
+{
+	int rc;
+	int size;
+	uint16_t offset = 0;
+	uint16_t max_offset;
+	struct coredump_hdr_t dump_hdr;
+	struct coredump_arch_hdr_t arch_hdr;
+	struct arm_arch_block arch_blk;
+	struct coredump_threads_meta_hdr_t thread_meta_hdr;
+	struct coredump_mem_hdr_t mem_hdr;
+	struct dump_metadata metadata;
+	uint8_t header_id;
+
+	rc = coredump_cmd(COREDUMP_CMD_VERIFY_STORED_DUMP, NULL);
+	if (rc == 0) {
+		return -ENOENT;
+	} else if (rc != 1) {
+		return rc;
+	}
+
+	size = coredump_query(COREDUMP_QUERY_GET_STORED_DUMP_SIZE, NULL);
+	if (rc <= 0) {
+		return rc;
+	}
+
+	max_offset = offset + size;
```

Comment:
```suggestion
	max_offset = size;
```
`offset` is always 0